### DISCUSSION TITLE
Update beyond 8.0.0.6, VdhCoApp 2.0.9 breaks on download (8.1+ goes directly to system dns disregarding browser settings)

### DISCUSSION THREAD

**Discussion by pompomwifi** [aclap-dev/video-downloadhelper#273] [Help and Q&A] [answered]
### Your question

Update1: [WORKAROUND]
I'll stay with workaround method that I typed below for the time being. Have absolutely no idea how any of the newer versions of extension and or companion app works for other ppl.

Update2: [CASE CLOSED/ANSWERED]
OK, thanks everyone, the culprit has been found! For whatever bizzare reasoning Video downloadhelper starting from version 8.1 and beyond decided to suddenly skip your chosen browser DNS over HTTPS connection setting and goes directly to your system DNS (or router).

In my infinite wisdom long long time ago, I dont know what possesed me, trully, but I had changed system DNS to the family friendly option of cloudflare. (1.1.1.3, 1.0.0.3)

Awkward, ashamed: Smart, me.

-----

Updating from older 7.X extension and its app to the curent ones available on stock win11 x64, firefox nightly and firefox stable, breaks download on the website mentioned below.

I tried to remove old companion app, reinstalled everything again, rebooted PC. Yet, still, nothing works: After pressing download button and waiting 1-2 seconds, there will always popup a red error msg about 'Exit code 1'.

Beta ver. of extension produced the same result. Tested with Firefox Nightly+Stable, Edge(default win11), Ungoogled Chromium portable - same result as follow:

```
Exit code: 1

moz-extension://3ff43a0e-7bdb-4c1c-be6b-534745d0e478/background/main.js:1
receive@moz-extension://3ff43a0e-7bdb-4c1c-be6b-534745d0e478/background/main.js:1:13157
Zp</callCatchAppNotFound/</<@moz-extension://3ff43a0e-7bdb-4c1c-be6b-534745d0e478/background/main.js:1:42626
```

### Addon & CoApp version and details

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\greenhatguy",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by GrampaWildWilly** [comment]
This is not my experience.  Do provide one URL or a few on which you are having trouble.  I can't test "every URL."  I can test only the ones you tell me about.

By the way, is Firefox 125 a nightly build?  Or is that just a typo.  You know, you can edit your own posts.  Look at the top of your post to the right.  Click on the 3 dots.  I think you'll be able to take it from there.

**Reply by pompomwifi** [reply]
Saying 'every URL', I meant literally any.

For example family friendly one:
https://www.youtube.com/watch?v=NfwSvGru-Ks
= failed

Or, open any top 10 porn sites, any links whatsoever, wont work for me; tried and failed to download any:
https://www.xvideos.com/
https://www.pornhub.com/

However I've found some workaround instead,
Once installed older 8.0.0.6 version with VdhCoApp 2.0.9, all I had to do is change some settings:
Behavior - Browser
Default download dir - E:\DLDS
Chunked data CoApp requests - Off
Check CoApp on startup - Off
Player Enabled - Off

With this config, extension works just fine, if a little barebones, loosing some advanced functions, but overall experience is better than any other alternative, at least for me. No errors with 8.0.0.6 + VdhCoApp 2.0.9.

8.0.0.6 + VdhCoApp 2.0.10 also works with some tweaking (and even less functionality):
Needs HLS as M2TS Enabled (Download HLS streams as M2TS) which is not something I personally want as sound quality suffers a lot (10kbps rather than 90kbps same video HLS if installed VdhCoApp 2.0.9 without M2TS Enabled); Any other (newer) versions of extension and app keep failing consistently.

Also, I have found that downgrading VdhCoApp to 2.0.9 / 2.0.10 from the current latest is possible only if the app is uninstalled first, then and only then installing prev. (older) version works correctly. This doesn't help with my problem (error exit code: 1) though. Just something I found worth mentioning about.

**Reply by GrampaWildWilly** [reply]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3235, Firefox 123.0.1 64-bit, licensed VDH 8.2.2.6a1 beta, CoApp 2.0.19.  I do not work for VDH.  I'm just another user.

For your YouTube example, here is what happened to me:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/93114f6d-cfe5-492c-93f6-dad5b3abecef)

I got these results:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/cc56dd9a-9845-4bf7-8c9e-d4cefdefbc6c)

Here is one of them playing in VLC:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7b3edf2b-afe8-437c-80f8-e415671a017e)

They were identical videos.  Both played all the way through, with video & audio.  I didn't actually sit & watch them, just sampled them at intervals.  They appeared to be fine.

So I have to say I can't replicate your results.

**Reply by GrampaWildWilly** [reply]
I have several examples elsewhere in this forum in which I successfully downloaded from Pornhub so I'm not going to do another one.

**Reply by GrampaWildWilly** [reply]
Since we need to keep this a G-rated forum, or something approximating that, I have included another series of images similar to the ones above for YouTube in the attached zip file.  I included the URL I picked at random off XVideos in a .txt file within the zip file.  Since I went with this approach to hide things, I didn't bother to censor the images at all.  The contents of the attached zip file are decidedly NSFW.  In fact, they are easily at least triple X rated.  You have been warned.

As you probably have guessed by now, my results on XVideos were no different from my results on YouTube.  I can't explain why you are having problems with this content.  I'm not.  If your evasive maneuvers work for you, I suppose you should stick with your setup.

[XV.zip](https://github.com/aclap-dev/video-downloadhelper/files/14544386/XV.zip)

Oh.  I just noticed I forgot to show the rest of the VDH menu for your YouTube video above.  Oh well.  No loss.

**Reply by pompomwifi** [reply]
Tried Youtube again after mutliple reinstalling of Video downloadhelper, its app and forcing any Windows services to Default. Absolutely clean win11 system. Result is that Youtube works again! Great, right? Not really, that's the only thing that start working, apparently. 

I followed your link to NSFW website, that page with 2 women and that's what I got on pressing any download under the extension menu:
![2024-03-09 05_35_02](https://github.com/aclap-dev/video-downloadhelper/assets/5382709/b0b6e333-9e13-40bd-8cac-4d6f0aa86448)

**Comment by Mattmjs** [comment]
The Mozilla site says it's the nightly build, I'm not sure the extension supports this build. Other than is it a anti virus or firewall issue ?
What folder / directory is the download going to ?

**Reply by pompomwifi** [reply]
I do undestand that any glitches, bugs and other problem on a Nightly build is not to be taken as legit reason to panic yet. However, as I mentioned above, this behavior is repeatable on any otehr browser, stable build. Just for this answer, I installed fresh Firefox stable -- same problem.

Files are being saved on a secondary drive NVME SSD as any other user files in my case. No problems with read and write access anywhere else. No was any problems before with downloading with pre 8.X ver.

Local Admin account;
UAC default mode;
MS Defender disabled;
BitDefender default settings;

winver.exe tells me I have Windows 11 Pro 64bit 23H2 22631.3155; It is a little modded though via powershell ( irm christitus.com/win | iex ) , so maybe that could be the reason? All it does is changing autorun-mode of some OS background services by putting them to manual run, rather than auto. Dunno why that could potentially break browser extension, but I'll try put them back and repeat the test. Nothing else is changed in the OS. Stock win11 x64.
Update: Nope. Same. Pretty much turned back time on any System wide changes, win11 x64 right now is as stock as it gets. Everything turned Default. Browser FF Stable.
Update2: Youtube start working again. After another reboot and reinstall. Alas, other websites prev. mentioned - do not. For Y2B I kinda always used 4k youtube downloader, so to me that's not good enough reason for updating from 8.0;

**Reply by Mattmjs** [reply]
Have you checked that your anti virus & firewall are not blocking / quarantining the companion app. Other uses reported that
BitDefender  was the culprit. You have to allow or whitelist companion app.

**Reply by GrampaWildWilly** [reply]
Yes, the easy thing to do is tell BitDefender antivirus that everything in the CoApp directory is permitted, similarly those same things are permitted through the BitDefender firewall.

BUT . . .

Downloads from YouTube apparently work.  That would say BitDefender is not getting in the way.  This is a most curious case.  Catching at straws here.  Maybe certain web sites are being blocked in the firewall.  Is that even possible?  I've never used BitDefender so I'm just throwing out ideas.

**Reply by GrampaWildWilly** [reply]
Changing certain services -- in the sense of Services in the Windows vernacular -- from autostart to start on demand seems utterly irrelevant.  I don't think that is affecting things.

**Reply by pompomwifi** [reply]
![2024-03-09 06_04_01-Error - Exit code_ 1, unable to download vids after update from 8 0 0 6, VdhCoAp](https://github.com/aclap-dev/video-downloadhelper/assets/5382709/96baab5e-51f0-4389-979e-b4e8f7b0d30d)

Added exceptiones a few minutes ago like u said. Disabled BitDefender for 1 hr. Then reinstalled companion app while FF was closed. Opened up FF Stable, - still nothing.

**Reply by GrampaWildWilly** [reply]
I've mentioned this before.  I sure wish VDH generated some sort of debugging log.  But there doesn't seem to be any instrumentation included with the publicly available versions of the browser extension & the CoApp.  I think such logs would be of immense help here.

**Reply by pompomwifi** [reply]
Free ver. of BitDefender in fact doesn't have Firewall, I'm using system's standard one instead.

**Reply by GrampaWildWilly** [reply]
OK.  That takes care of the antivirus side of things.  What about the firewall?  They are, as far as I've been able to understand, separate & independent parts of BitDefender.

**Reply by GrampaWildWilly** [reply]
Oh.  OK.  Well then I don't understand.  I'm using plain old Windows Defender for both antivirus & firewall.  It's never made a peep on the CoApp.

**Reply by GrampaWildWilly** [reply]
More catching at straws.  Let's see your first page of VDH settings with the sections expanded . . . except for the license section.

**Reply by Mattmjs** [reply]
> Files are being saved on a secondary drive NVME SSD as any other user files in my case. No problems with read and write access anywhere else. No was any problems before with downloading with pre 8.X ver.

What about just downloading on internal drive that Windows is installed on. Is there failure occurring there ?

**Reply by pompomwifi** [reply, marked as answer]
OK, thanks everyone, the culprit has been found! For whatever bizzare reasoning Video downloadhelper starting from version 8.1 and beyond decided to suddenly skip your chosen browser DNS over HTTPS connection setting and goes directly to your system DNS (or router).

In my infinite wisdom long long time ago, I dont know what possesed me, trully, but I had changed system DNS to the family friendly option of cloudflare. (1.1.1.3, 1.0.0.3)

_Awkward, ashamed:_ Smart, me.

My steps if anyone interested:
removed system DNS 'cloudflare family friendly' from every active connection,
disabled DNS over HTTPS on every browser,
disconnected Lan / WiFi connections in windows setting,
opened CMD as admin, typed as follow:
ipconfig /flushdns
started LAN connection again,
gone though cmd's ipconfig /flushdns command again for good measure,
didn't bother with WiFi for clean test,
and, finally, finally, it works!!

**Reply by GrampaWildWilly** [reply]
I'm glad you found it because I've never messed with any of that stuff.  After reading your solution I went into the Firefox settings & started scrounging around.  In the Privacy & Security section there's a section with the title, "Enable DNS over HTTPS using" all the way at the bottom.  I don't claim to entirely understand it but I've never changed it from the default selection.  Are you saying you've changed that?  In any case, I suppose this one goes in the tool box as another possible issue, alongside the antivirus.

Speaking of the Firefox settings, they are pathological in their user friendliness.  They are quite user unfriendly.  Even after double clicking some of the text to try to put focus onto the page, the arrow keys & page keys do not scroll the page of settings.  It goes to another section of settings instead.  Most annoying.

**Reply by Mattmjs** [reply]
Well guess what, I enabled DNS over HTTPS using Cloudflare. I checked it was using  cloudflare at dnsleaktest.com
I tried your youtube example and another download from xvideos, both worked.

Maybe the DNS over HTTPS in your browser stopped working and therefore used the system dns.

**Reply by GrampaWildWilly** [reply]
There's just something that doesn't add up about this DNS thing.  @pompomwifi says he could visit the sites & even play the content.  But VDH wouldn't download it.  I don't understand how any DNS setting would let you visit the web page & play content off it but not let you download from it.  Unless . . . the streaming for playback comes from one site & the download comes from another.  It sounds like there were parental controls in place for cloudflare, which evidently provides the download.  But whatever supplies the web site & in-page streaming is some other site on which there were no parental controls.  Now, I have to admit I'm just making up scenarios to fit the observed behavior.  I'm guessing.  But I just don't understand what was going on here.

**Reply by strx5** [reply]
So, VDH now ignores 'DNS over HTTPS' settings in Firefox, right? It just "defaults" to whatever DNS provider your system relies upon? Do I need to add/change my DNS provider (i.e., tweak the Internet connection config of my router) to avoid this ridiculous error?

**Reply by Mattmjs** [reply]
@strx5 , it's not the case that VDH will ignore dns settings in Firefox. pompomwifi said it did that and used the system dns
which was a family friendly dns by Cloudflare. I don't believe so because how did they visit those porn sites ?
That version of Cloudflare dns does not resolve porn sites. The fact they were still able to suggests it was working in Firefox.

There are odd cases where dns over https will fail, I saw that one time using nextdns. You don't need to worry about changing
anything. If it fails on rare occasions like I said, it'll will revert to the system dns. That's not VDH ignoring or overriding the dns.

**Comment by Mattmjs** [comment]
I don't understand why it behaved like that either, I could try replicating
it to see what happens but computer is off.
It's too hot to run it, it was around 38 Celsius today  & another 2 days
forecast in the high thirties. reply via Gmail app.

**Reply by Mattmjs** [reply]
> cloudflare. (1.1.1.3, 1.0.0.3)

With that dns running it blocks known malware and adult content addresses. It doesn't even resolve those porn domains above.

![Screenshot 2024-03-10 120838](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/e209818b-ca9a-43a1-96d8-2e2121b6769c)

**Reply by GrampaWildWilly** [reply]
Whoa . . . Now that is hot.  Mind you, I think my system runs at something approximating that temperature.  That would be the internal temperature of the computer, not the ambient temperature of the room.  I didn't set out to do things this way for the purposes of cooling things.  But I ended up with 8 fans in my new system:
- 1 fan in the power supply
- 2 fans in the CPU cooler
- 2 fans in the front of the case
- 3 fans in my video adapter

I wonder if my system would care about weather that ridiculously hot.  I assume you're working from experience when you shut off your computer when it gets that hot out.  But it seems to me that I've seen my BIOS reporting an operating temperature in the 30s C.  38C seems like it's normal operating temperature for the computer.  But I'm not in Canberra in southern hemisphere summer so I'm just speculating.  Here, right now, it's about +5C, but that's at 21:00, well after sundown.

**Reply by GrampaWildWilly** [reply]
I usually don't complain about the weather.  I say if you don't have to shovel it, it's fine.  Rain, heat, whatever.  Only snow gets me to grumble.

**Reply by Mattmjs** [reply]
It can get hot in Canberra too in the summer but I'm down in Melbourne.

**Reply by GrampaWildWilly** [reply]
Oops. 😺
